### ISSUE TITLE
Support for Angular ESlint Template Parser

### ISSUE THREAD

**Issue by RaphaelHaettich** [schoero/eslint-plugin-readable-tailwind#76]
It would be awesome if you could add support for Angular with the [@angular-eslint/template-parser](https://www.npmjs.com/package/@angular-eslint/template-parser).
Currently Angular uses its own parser instead of the @html-eslint/parser for HTML files, so this plugin sadly does not work with it. 
Logically it should not differ much from the HTML version execpt possibility to use [class]="'p-4'" and the use of {{}} in the templates.

**Comment by schoero** [comment]
I'm not familiar with Angular, but I did try adding Angular support a while ago. However, I found it quite overwhelming at the time.

Please correct me if I'm wrong, but now that I've revisited it, it seems there are many different ways to apply (conditional) classes in Angular.

```html
<div class="text-3xl font-bold mb-4">
  Normal classes
</div>
```

```html
<div class="{{ isActive ? 'text-green-500' : 'text-red-500' }}">
  Interpolated class attribute
</div>
```

```html
<div [class.text-green-500]="isActive" [class.text-red-500]="!isActive">
  Single class binding (this can be ignored for the plugin)
</div>
```

```html
<div [class]="['font-bold', isActive ? 'text-green-500' : 'text-red-500']">
  Array binding using [class]
</div>
```

```html
<div [ngClass]="{ 'text-green-500': isActive, 'text-red-500': !isActive }">
  Object syntax with [ngClass]
</div>
```

```html
<div [ngClass]="['font-bold', isActive ? 'text-green-500' : 'text-red-500' ]">
  Array syntax with [ngClass]
</div>
```

```html
<div [ngClass]="'text-blue-500 font-bold'">
  String syntax with [ngClass]
</div>
```

And the same applies to the templates where everything can be inside ``.

I'm not sure how easy it will be to support all different cases but I can try to look into it.

**Comment by mtpc** [comment]
#77 adds "normal classes" support, I still haven't worked on class binding or object syntax as it would probably take too much of my time unfortunately

**Comment by schoero** [comment]
I have implemented angular support in version 2.1.0. The angular parser is very different than all other parsers and I had to create many workarounds so angular support might be considered experimental right now. 

Check the [docs](https://github.com/schoero/eslint-plugin-readable-tailwind/blob/v2.1.0/docs/parsers/angular.md) to see how to integrate it in your eslint config.

Please report any issues you might find.

**Comment by schoero** [comment]
I'm going to close this issue as it is now resolved. If you encounter any problems with the angular implementation please open a new issue.
